### DISCUSSION TITLE
image-resize script broken?

### DISCUSSION THREAD

**Discussion by cfritze64** [johnlindquist/kit#626] [Errors]
The image-resize script located on the scriptkit homepage:
Used to work. Monterey broke it, or something else? I select a file in the finder, run the script, it accepts my input for a new width, then bails. 

the error log reads:
```
[2021-11-28 18:51:28.797] [warn]  ☠️ ERROR PROMPT SHOULD SHOW ☠️
[2021-11-28 18:51:28.802] [warn]  Error: Unsupported input '' of type string
    at Sharp._createInputDescriptor (/Users/BikeFixe/.kenv/node_modules/sharp/lib/input.js:54:11)
    at new Sharp (/Users/BikeFixe/.kenv/node_modules/sharp/lib/constructor.js:303:29)
    at Sharp (/Users/BikeFixe/.kenv/node_modules/sharp/lib/constructor.js:161:12)
    at file:///Users/BikeFixe/.kenv/scripts/image-resize.js:13:22
    at processTicksAndRejections (node:internal/process/task_queues:94:5)
```

**Comment by johnlindquist** [comment]
Hrrrrmmmmm.... 🤔

For some reason, I have no idea why, some commands no longer work "inline" (while some still do...):

```bash
osascript -e "tell foo bar baz"
```

## Workaround

I'll have `kit` create a `latest.scpt` file with the script contents, then have it run the script.

Kit writes these contents to `latest.scpt` in the script's tmp dir:

```osascript
foo bar baz
```

Then kit can execute the script:

```bash
osascript ~/.kenv/tmp/script-name/latest.scpt
```

This solution should be backwards compatible with all osx versions. No API changes will be required, so it should "just work" again after I push out another build.

Thanks for letting me know!

**Reply by cfritze64** [reply]
John:

Could it be related to [this issue](https://talk.macpowerusers.com/t/applescript-does-not-function-correctly-after-upgrade-to-macos-12/26259)? Permissions windows often get lost (or maybe don’t reliably show in the first place). I’m not running beta 116 of ScriptKit so I can’t go back to look what the permissions settings were in System Pref back when I encountered the problem with the image sizing script.

**Reply by johnlindquist** [reply]
Appreciate the follow-up. I don't think it's a permissions issue. 

All the other AppleScript helpers worked fine inline. This helper worked fine as a script, but not inline.

Honestly, I'm totally happy with the workaround, it fixes everything and gives you a nice little log of the previous applescript you ran.
